File: Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        vector<int>ans(nums.size());
        int j=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=0){
                ans[j] =nums[i];
                j++;
            }
        }
        nums = ans ;
      
    }
};

File: phase2/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        next_nonzero = 0

        for n  in nums:
            if n!=0:
                nums[next_nonzero] =n
                next_nonzero+=1
        print(next_nonzero)
        for i in range(next_nonzero,len(nums)):
            nums[i]=0
        
        